### PR TITLE
ci: rework smoke tests

### PR DESCRIPTION
**Pull request by m-ildefons**
- Make the smoke tests script more understandable by utilizing appropriately named functions to indicate expected results of operations
- Apply shellscript best-practices (e.g. quoting) to avoid unintended side-effects
- Avoid early error exit to avoid missing half the tests in case of error
- Make output user-friendly with a clear indication of success/failure

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

### CODE REVIEW

File: qa/rgw/store/sfs/tests/sfs-smoke-test.sh [hunk 1]
```diff
@@ -44,101 +41,158 @@ signature_v2 = True
...
+# it will ignore a failure, but not fail on success, unlike expect_failure,
+# which will fail on success
+# shellcheck disable=SC2317
+ignore_failure() {
+  "$@" || true
+}
+
+expect_failure() {
+  "$@" > /dev/null 2>&1 && failure
+  success
```
Comment (jecluis) [inline qa/rgw/store/sfs/tests/sfs-smoke-test.sh]: aren't we always outputting `success` here anyways, even if `failure` is hit?

File: qa/rgw/store/sfs/tests/sfs-smoke-test.sh [hunk 2]
```diff
@@ -44,101 +41,158 @@ signature_v2 = True
...
+}
+
+expect_success() {
+  "$@" > /dev/null 2>&1 || failure
+  success
```
Comment (jecluis) [inline qa/rgw/store/sfs/tests/sfs-smoke-test.sh]: and same here?

**Review by jecluis — DISMISSED** [review]
it looks good to me. I'm confused about a few things, commented on them, but I'm assuming it is behaving as it's supposed and that I'm just not grasping what is going on.

File: qa/rgw/store/sfs/tests/sfs-smoke-test.sh (hunk 1 above)
Comment (m-ildefons) [reply qa/rgw/store/sfs/tests/sfs-smoke-test.sh]: Yes. Good catch, this logic is fubar.

File: qa/rgw/store/sfs/tests/sfs-smoke-test.sh (hunk 2 above)
Comment (m-ildefons) [reply qa/rgw/store/sfs/tests/sfs-smoke-test.sh]: Yup, this is also wrong. Good catch

**Review by jecluis — APPROVED** [review]
lgtm

**Pull request merged**